### ISSUE TITLE
n/no-sync does not work with named import

### ISSUE THREAD

**Issue by blephy** [eslint-community/eslint-plugin-n#102]
When `'n/no-sync': ['error', { allowAtRootLevel: false }],`

```ts
import { readFileSync } from 'node:fs';

const TOTO: Buffer = readFileSync('/toto');
```

works

but 

```ts
import fs from 'node:fs';

const TOTO: Buffer = fs.readFileSync('/toto');
```

is reported as espected

**Comment by scagood** [comment]
mmm, there are a couple of ways to go about this 🤔

The main issue is that we only check for a MemberExpression (`something.else`) as long as one of the properties ends with `Sync`. (regardless of if they're called as a function)

eg:
```js
// These match
fs.magicSync('path')
fs.readFileSync.call(null, 'path')

// These do not match
readFileSync('path')
readFileSync.call(null, 'path')
```

Given that, we can just change the selector a little bit ([lib/rules/no-sync.js:36](https://github.com/eslint-community/eslint-plugin-n/blob/master/lib/rules/no-sync.js#L36-L38))
eg:
```js
[
  // fs.readFileSync()
  "MemberExpression[property.name=/.*Sync$/]",
  // readFileSync.call(null, 'path')
  "MemberExpression[object.name=/.*Sync$/]",
  // readFileSync()
  ":not(MemberExpression) > Identifier[name=/Sync$/]",
]
```

The other option is to try to detect the "Sync" imports from the built in imports.
Also, if I remember correctly `createReadStream` creates the `filehandle` synchronously, but that does not end with `Sync` so we may also need to have other functions marked as 'sync' (don't hold me to that though 😂)

This would be done using the import/require declarations, then we'd traverse to the variable usage to see whether a Sync method was used. I created a quick gist here:
https://gist.github.com/scagood/bb6513bfe84a35f682c604f27a59e775

**Comment by aladdin-add** [comment]
fixed by #127  and released in `16.2.0`.
